packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -52,39 +52,28 @@ export const createCart = payload =>
     };
 
 export const addItemToCart = (payload = {}) => {
-    const { item, fetchCartId } = payload;
+    const { addItemMutation, fetchCartId, item, quantity, parentSku } = payload;
+
     const writingImageToCache = writeImageToCache(item);
 
     return async function thunk(dispatch, getState) {
         await writingImageToCache;
         dispatch(actions.addItem.request(payload));
 
         try {
-            const { cart, user } = getState();
-            const { isSignedIn } = user;
-            let cartEndpoint;
-
-            if (!isSignedIn) {
-                const { cartId } = cart;
-
-                if (!cartId) {
-                    const missingCartIdError = new Error(
-                        'Missing required information: cartId'
-                    );
-                    missingCartIdError.noCartId = true;
-                    throw missingCartIdError;
-                }
-
-                cartEndpoint = `/rest/V1/guest-carts/${cartId}/items`;
-            } else {
-                cartEndpoint = '/rest/V1/carts/mine/items';
-            }
-
-            const quoteId = getQuoteIdForRest(cart, user);
-            const cartItem = toRESTCartItem(quoteId, payload);
-            await request(cartEndpoint, {
-                method: 'POST',
-                body: JSON.stringify({ cartItem })
+            const { cart } = getState();
+            const { cartId } = cart;
```

Comment:
So this can be `null` after an order is placed, and a subsequent `add` calls this with a `null` cartId.

@tjwiebell suggested two options, but I think it may be better to figure out why the cart is deleted but never re-created after order. It should definitely be created _every_ time the cartId is deleted.